### DISCUSSION TITLE
Character auto-step on surface

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#217] [Q&A] [answered]
I am trying to prevent the character auto-stepping onto an obstacle while it is in the air. Is there a way to prevent it doing it, if the char is not supported, e.g. while jumping? On the video, the jump should take me higher than the box. Instead, I am auto-landing on to its top surface. I think what happens is that the char is in the range of step up, and gets projected onto the surface. I'd like to disable that if the character is not supported.

https://github.com/user-attachments/assets/856e174a-a029-4477-966a-0856f66264f7

**Comment by jrouwe** [comment, marked as answer]
If you're using `CharacterVirtual::ExtendedUpdate` then you could set `ExtendedUpdateSettings::mWalkStairsStepUp` to zero while you're in air to disable stair walking.

**Reply by LeXXik** [reply]
That was an easy one. Thank you!

**Comment by LeXXik** [comment]
Hmm, I am making a new char controller, and faced the same issue. However, both `mWalkStairsStepUp` and `mStickToFloorStepDown` are set to zero in the extended update settings, regardless if the char is in the air or not. I also changed the shape to box, to make sure no capsule curvature is in play here. The char settings are mostly the same as in the default example. Any idea what may cause it?


https://github.com/user-attachments/assets/298a673e-a500-4421-8383-e611b84fd018

**Reply by jrouwe** [reply]
No, I really don't see how that would be possible without stair stepping.
